### PR TITLE
CN User Stats Page Load Test Data

### PR DESCRIPTION
**Pull request by chrisn02**
Have made it so the users' stats are displayed on this page. There is a graph that displays the percentage starting from their lowest score to their highest score. Then there are two boxes that shows the total points and average percentage. These are displayed using icons, but if you hover each one it will say "Total Points" and "Average Score" respectively. There is then a table that shows information about each test. There is pagination in the table which limits 10 to be displayed at a time, which was done using DataTables.

image

### CODE REVIEW

File: user-stats.php [hunk 1]
```diff
@@ -1,12 +1,79 @@
...
+
+    // Get percentages
+    $stmt = $conn->prepare("CALL GetStudentsQuizPercentages(:studentID)");
+    $stmt->bindValue(":studentID", $studentID, PDO::PARAM_STR);
+    $stmt->execute();
+    $percentages = $stmt->fetchAll(PDO::FETCH_COLUMN, 0); // Fetch only the first column of each row
+
+    header('Content-Type: application/json');
+    echo json_encode(['testNames' => $testNames, 'percentages' => $percentages]);
+    exit;
```
Comment (Jforjo) [inline user-stats.php:47]: You can put the echo content inside the exit, like 'exit(json_encode...)'

**Review by Jforjo — APPROVED** [review]
Looks good, although you can add the nav bar in so the user is able to go back to the dashboard or logout.

**Comment by chrisn02** [comment]
> Looks good, although you can add the nav bar in so the user is able to go back to the dashboard or logout.

Thanks, there is a navbar there but it was just cut off from the screenshot

**Review by gchodancar19 — APPROVED** [review]
It looks to me but just the arrow needs to be a bit bigger.

**Pull request merged**